### PR TITLE
[Feature 6] [Feature 6.1] Selection System + Selection screen switch

### LINKED ISSUE #8: [Feature 6] Selecting and moving Files within directory
Allow the user to select files and move them to different folders within the file management system.

### LINKED ISSUE #9: [Feature 6.1] Selection screen and switch to selection mode
The view associated with the selection mode, and the process of switching to the selection mode(internal changes).

### LINKED ISSUE #29: [Feature 6.2] Selection and notify user
Allow the user to select files, and display that the files have been selected.

### PR DESCRIPTION
**Pull request by ChowdhuryShaswata**
Implemented a file selection system accessed through the Select icon in the directory toolbar. Functions intuitively: clicking on files selects and highlights them, clicking on already selected files removes the highlight and removes the files from the list. Longclick accesses a filemenu allowing same functionality as in the regular directory. Clicking the select icon returns to the regular directory access mode and clears the list of selected files.

SelectionController: Counterpart to DirectoryAccessController, binds view elements and actions on the selection screeen (including toolbar). Binds SelectionListAdapter to the recyclerview element.

SelectionListAdapter: Alternative click and longclick mapping for holder items displayed in recyclerview. Click: Select/Deselect, Longclick adjusted to redirect to Selection classes rather than directory classes where appropriate.

SelectionMenuController: Binds the selection menu interactor to the file menu. Extends fileMenuInteractor for duplicate functionality otherwise.

SelectionMenuInteractor: Extends FileMenuInteractor. Performs selection menu actions, which are the same as directory filemenu actions, other than overriding open() to send the selected list of files to the new selection fragment generated.

SelectionInteractor: Implements SelectionInputBoundary.

### CODE REVIEW

File: app/src/main/java/com/example/notes2text/adapters/fragments/DirectoryAccessController.java [hunk 1]
```diff
@@ -22,9 +34,13 @@
...
 
+    //Required collaborators
+    private DirectoryAccessOutputBoundary directoryPresenter = new DirectoryAccessPresenter();
+
+
     // TODO: Rename parameter arguments, choose names that match
     // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-    private static final String ARG_PARAM1 = "param1";
+    private static final String ARG_FILE_PATH = "param1";
     private static final String ARG_PARAM2 = "param2";
```
Comment (CSEndeavour) [inline app/src/main/java/com/example/notes2text/adapters/fragments/DirectoryAccessController.java:48]: You can remove this variable if it's not needed.

**Review by CSEndeavour — APPROVED** [review]

**Comment by CSEndeavour** [comment]
Add docstrings to your code.

**Comment by CSEndeavour** [comment]
Create test cases for SelectionInteractor and SelectionPresenter.

File: app/src/main/java/com/example/notes2text/fileselection/adapters/SelectionController.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    private void setToolbarMenu() {
+        requireActivity().addMenuProvider(new MenuProvider() {
+            @Override
+            public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater menuInflater) {
+                //Inflate the menu
+                menuInflater.inflate(R.menu.selection_toolbar, menu);
+            }
+
+            @Override
+            public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
```
Comment (JerryLi-617) [inline app/src/main/java/com/example/notes2text/fileselection/adapters/SelectionController.java:168]: The return value seems to be unused, perhaps there are other usage of it or more suitable methods?

File: app/build.gradle [hunk 3]
```diff
@@ -10,7 +10,7 @@ android {
         applicationId "com.example.notes2text"
         minSdk 23
         //noinspection ExpiredTargetSdkVersion
-        targetSdk 30
+        targetSdk 29
```
Comment (ChowdhuryShaswata) [inline app/build.gradle:13]: targetsdk rolled back to allow non-image files to be detectable.

File: app/src/main/java/com/example/notes2text/fileselection/usecases/SelectionMenuInteractor.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import com.example.notes2text.adapters.ActivitySwitchController;
+import com.example.notes2text.fileselection.adapters.SelectionController;
+import com.example.notes2text.usecases.FileMenuInteractor;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class SelectionMenuInteractor extends FileMenuInteractor {
+
+    ArrayList<File> selectedList;
```
Comment (JerryLi-617) [inline app/src/main/java/com/example/notes2text/fileselection/usecases/SelectionMenuInteractor.java:19]: Consider making this variable private

**Review by JerryLi-617 — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/adapters/SelectionListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;;import com.example.notes2text.entities.FileViewHolder;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class SelectionListAdapter extends FileListAdaptor {
+
+    private ArrayList<File> selectedFiles;
```
Comment (CSEndeavour) [inline app/src/main/java/com/example/notes2text/adapters/SelectionListAdapter.java]: This variable isn't accessed anywhere so it can be removed.

**Pull request merged**